client/verta/verta/registry/errors.py

```diff
@@ -0,0 +1,18 @@
+class CustomModelError(Exception):
+    """Exception to be raised in user-defined models.
+
+    Messages of this class are caught in model containers and converted to
+    HTTP errors with the specified HTTP code and message in the response body.
+
+    Attributes
+    ----------
+    http_code : int
+        The http code to be returned by a wrapping service.
+    message : str
+        The response message.
+    """
+
+    def __init__(self, http_code, message):
+        self.http_code = http_code
+        self.message = message
```

Comment:
I wonder if this could be a `property` for `self.args[0]`, but that might be an over-optimization.

Reply:
Let's keep it simple.

Reply:
I'm unsure whether having three ways to obtain the message string from this exception class (`e.message`, `str(e)`, and `e.args[0]`), with this one even being mutable, is simpler than just the two that `Exception` offers.

But this is somewhat beyond my familiarity with Python conventions so I'm alright with revisiting it later if it becomes any sort of real issue 😆